### ISSUE TITLE
Firefox Mint-Y-Dark and dark themes in general Bug + Fix

### ISSUE THREAD

**Issue by smurphos** [linuxmint/mint-19.1-beta#30]
**Describe the bug**
Longstanding readability issues with firefox content widget display in using Mint-Y-Dark and any other dark theme

**To Reproduce**
Set Mint-Y-Dark as GTK / Controls theme - experience text readability issues in various Firefox content widgets - particularly entries.

**Expected behavior**
No readability issues.

**Frequency**
Does the issue happen:
- [x ] Always
- [ ] Randomly

**Edition (Desktop environment):**
In which edition is this happening?
- [ x] Cinnamon Edition
- [ x] MATE Edition
- [ x] Xfce Edition

**Live or post-install:**
Indicate if this is happening after the installation, or during the live session:
- [x] Post-install
- [x] Live session

**The Fix**

Ship the Mint Firefox build with an override in `/usr/lib/firefox/distribution/distribution.ini` to create and set this key in `about:config`

`widget.content.gtk-theme-override` with a value of `Mint-Y`

**Comment by JosephMcc** [comment]
Personally I'm not sure this is a great idea. It would be unclear to most users why Firefox isn't adapting to their theme changes.

**Comment by smurphos** [comment]
It doesn't effect the application of the GTK theme to the firefox header bar / menus etc only to the elements of website content that firefox partially themes.

**Comment by JosephMcc** [comment]
Hmm... I didn't realize that. I'd have to play with it a bit. It would raise bug reports from people who choose say the blue version and get stuck with green highlights, correct?

**Comment by smurphos** [comment]
Possibly or you could just go with Adwaita and be done with it.  I don't really understand why users would want website content to be themed by the GTK theme anyway. Chromium based browsers don't do it.

**Comment by gm10** [comment]
Personally I have it set to

`user_pref("widget.content.gtk-theme-override", "#");`

That (or any other invalid but not empty name) makes it fall back to its default, un-themed controls (at least on MATE). Seems preferable to me to using a specific theme. 

Or just wait for Mozilla to fix this, the bug report on this has only been open for 18 years now. :100:

**Comment by smurphos** [comment]
user_pref("widget.content.gtk-theme-override", "#"); works well...
![screenshot-screen-2018-12-08-164936](https://user-images.githubusercontent.com/29017677/49688407-58ecee00-fb09-11e8-81d7-d32eca3cf3f1.png)

**Comment by NikoKrause** [comment]
So, do I get it right, that if you choose a Dark variant for Firefox, e.g. `Adwaita-Dark` or `Mint-Y-Dark-Red`, you want Firefox to use/gtk-theme-override to `Adwaita` or `Mint-Y-Red`.

At least in Cinnamon it's possible to do this on theme change.
We would just need to create or change the file `/home/USER/.mozilla/firefox/***.default/user.js` with the content:
```
user_pref("widget.content.gtk-theme-override", "My-Theme-without-Darkl");
```

**Comment by gm10** [comment]
@NikoKrause that doesn't seem a reliable solution. Not all dark themes have a convenient `-Dark` suffix. The global override seems far more reliable, and also you're not taking the decision away from the user that way by simply modifying their local profile.

**Comment by smurphos** [comment]
Agreed with @gm10 - I think users would be happiest to have their web content delivered in a consistent readable fashion as intended by the web site designer without interference from their choice of system theme.

**Comment by clefebvre** [comment]
Fixed in ubuntu-system-adjustments 2018.12.12.
